ballerina/tests/listener_tests.bal

```diff
@@ -153,3 +153,13 @@ function testListenerAttachDetatch() returns error? {
     check logServer.attach(dummyService);
     check logServer.detach(dummyService);
 }
+
+@test:Config {}
+function testReadOnly() returns error? {
+    ConnectClient socketClient = check new ("localhost", 9005);
+    string msg = "Echo from connect client";
+    check socketClient->writeBytes(msg.toBytes());
+    readonly & byte[] response = check socketClient->readBytes();
```

Comment:
Shall we add a assertion here to see whether `response` is `readonly` ?

Reply:
That is what I have done in line 163. 
The server will respond with the result of `data is readonly & byte[]` (whether this is true or false) in the `onBytes` function. It is asserted in the next line.
``test:assertEquals(string:fromBytes(response), "true");``